### PR TITLE
Remove SipHash

### PR DESCRIPTION
**Pull request by michaelsproul**

### CODE REVIEW

**Comment by michaelsproul** [comment]
Plan of action, to be implemented by me, soon...

* Don't put `sha3` in this crate (yet), but *do* remove SipHash (no urgency to re-release).
* Put the necessary `sha3` stuff in `routing` for now, and depend on that from `safe_vault`.
* Later: assess hash function usage and consider using SHA2/3 everywhere with multihash.

**Comment by ustulation** [comment]
@michaelsproul So i guess this is no longer needed - could u pls submit a PR then to get rid of sip-hasher thingy from utilities ? You can close this one if you want or make this the PR with the changes to just remove sip hasher.

**Comment by michaelsproul** [comment]
I've updated this PR to just remove SipHash, the only remaining mention of sha3 is the branch name.

File: src/lib.rs [hunk 1]
```diff
@@ -48,7 +48,7 @@ extern crate bincode;
 extern crate config_file_handler;
 #[macro_use]
 extern crate lazy_static;
-#[macro_use]
+#[cfg_attr(test, macro_use)]
```
Comment (ustulation) [inline src/lib.rs]: Why is this change needed ? (Is it something to do with the treatment of macros in newer rusts ? )

(On a side note: Pls rebase against latest master - should be good to merge after that)

**Review by ustulation — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (michaelsproul) [reply src/lib.rs]: Yeah, I've been trying out Rust's new incremental compilation, and the new compiler is smart enough to know that the macros are only used in tests.

Comment (michaelsproul) [reply src/lib.rs]: (rebased onto master)

Comment (michaelsproul) [reply src/lib.rs]: ready to go @ustulation, although safe_vault will break if we release a new version without merging https://github.com/maidsafe/safe_vault/pull/606 (which depends on routing master, or a new version of routing)

**Review by ustulation — APPROVED** [review]

**Pull request merged**